<a href="https://colab.research.google.com/github/kaneserignembacke260-beep/titanic-ml-pipeline/blob/main/05_mod%C3%A9lisation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [28]:
# On importe pandas pour manipuler les données
import pandas as pd

# On charge le fichier CSV déjà nettoyé (créé dans le notebook 04_nettoyage)
df = pd.read_csv("titanic_clean.csv")

# On vérifie que tout s'est bien chargé
df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    int64  
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  FamilySize   891 non-null    int64  
 11  Title        890 non-null    object 
 12  Embarked_Q   891 non-null    bool   
 13  Embarked_S   891 non-null    bool   
dtypes: bool(2), float64(2), int64(7), object(3)
memory usage: 85.4+ KB


In [29]:
# On choisit les colonnes qui vont servir d'indices pour le modèle
features = ["Pclass", "Sex", "Age", "FamilySize"]

# X = variables explicatives (les indices)
X = df[features]

# y = variable cible (ce qu'on veut prédire)
y = df["Survived"]

# On vérifie les dimensions
print(X.shape)
print(y.shape)

(891, 4)
(891,)


In [30]:
# On importe la fonction qui permet de séparer les données
from sklearn.model_selection import train_test_split

# On sépare en 80% entraînement / 20% test
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# On vérifie les tailles
print("Taille train :", X_train.shape)
print("Taille test :", X_test.shape)

Taille train : (712, 4)
Taille test : (179, 4)


In [31]:
# On importe le modèle de régression logistique
from sklearn.linear_model import LogisticRegression

# On crée le modèle (pas encore entraîné à ce stade)
model = LogisticRegression()

# On entraîne le modèle UNIQUEMENT sur les données d'entraînement
model.fit(X_train, y_train)

LogisticRegression()

In [32]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [33]:
y_pred = model.predict(X_test)



- `y_pred` = prédictions du modèle sur les passagers du **jeu de test** (jamais vus pendant l'entraînement)
- `y_test` = les vraies réponses (ce qui s'est réellement passé)
- **Accuracy** = % de prédictions correctes → mesure la performance globale du modèle

In [34]:
accuracy=accuracy_score(y_test, y_pred)
print(accuracy)

0.8044692737430168


In [35]:
print(confusion_matrix(y_test, y_pred))

[[90 15]
 [20 54]]


 ce resultat  représente  ce tableau là
 	## 🎯 Matrice de confusion — Jour 12

|                        | Prédit : 0 (Mort) | Prédit : 1 (Survécu) |
|------------------------|:---:|:---:|
| **Réel : 0 (Mort)**    | 90 | 15 |
| **Réel : 1 (Survécu)** | 20 | 54 |

- **90** vrais négatifs (mort prédit, mort réel) ✅
- **54** vrais positifs (survécu prédit, survécu réel) ✅
- **15** faux positifs (survécu prédit, mort réel) ❌
- **20** faux négatifs (mort prédit, survécu réel) ❌

In [36]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.82      0.86      0.84       105
           1       0.78      0.73      0.76        74

    accuracy                           0.80       179
   macro avg       0.80      0.79      0.80       179
weighted avg       0.80      0.80      0.80       179



In [37]:
 from sklearn.ensemble import RandomForestClassifier

In [38]:
rf_model=RandomForestClassifier(n_estimators=100, random_state=42)

In [39]:
rf_model.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [40]:
y_pred_rf=rf_model.predict(X_test)

In [41]:
print(accuracy_score(y_test, y_pred_rf))

0.8156424581005587


In [42]:
print(confusion_matrix(y_test, y_pred_rf))

[[91 14]
 [19 55]]


In [43]:
print(classification_report(y_test, y_pred_rf))

              precision    recall  f1-score   support

           0       0.83      0.87      0.85       105
           1       0.80      0.74      0.77        74

    accuracy                           0.82       179
   macro avg       0.81      0.80      0.81       179
weighted avg       0.81      0.82      0.81       179



## 🎯 Comparaison des modèles — Jour 13

| Modèle | Accuracy |
|---|:---:|
| Logistic Regression (baseline) | 0.80 |
| Random Forest | 0.82 |

**Conclusion :** le Random Forest surpasse légèrement la régression logistique sur toutes les métriques (accuracy, precision, recall, f1-score). Amélioration modeste mais cohérente — logique avec seulement 4 features utilisées.